src/App.tsx

```diff
@@ -17,18 +37,22 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todosInitial={todosInitial}
+                todosToRender={setTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 && <Loader />}
+
+              <TodoList todos={todos} setCurrentUserId={setUserId} />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalShowed === true && <TodoModal user={user} todos={todos} />}
```

Comment:
```suggestion
      {selectedTodo && <TodoModal todo={selectedTodo} onClose={() => setSelectedTodo(null)} />}
```
take the userId from the todo and fetch the user inside the TodoModal